spring-cloud-contract-wiremock/src/test/java/org/springframework/cloud/contract/wiremock/AutoConfigureWireMockRandomPortApplicationTests.java

```diff
@@ -38,11 +38,21 @@
 	@Autowired
 	private Service service;
 
+	@Autowired
+	private WireMockProperties wireMockProperties;
+
 	@Test
 	public void contextLoads() throws Exception {
 		stubFor(get(urlEqualTo("/test")).willReturn(aResponse()
 				.withHeader("Content-Type", "text/plain").withBody("Hello World!")));
 		assertThat(this.service.go()).isEqualTo("Hello World!");
 	}
 
+	@Test
+	public void contextIsNotReload() throws Exception {
```

Comment:
Maybe change test method name to something like `dynamicPortsAreResolvedCorrectly`? You don't really assert here if the context is reloaded.

Reply:
I've updated the code as directed. Thanks for coming back to me so quick.

Reply:
Sure. However, I see the build is failing exactly with the issue that checking ports was introduced to fix, so it looks like, there has been no context cleanup where it needed to happen.

Reply:
Sorry about that. Late night code changes without applicable tests never workout well. I've fixed the logic around portIsFixed.

The tests were failing as previous classes weren't being marked to have the Spring context reloaded. The code worked before the change as it always reloaded the Spring context. I don't think it's a good idea that a test can pass in isolation but fail when run as part of a suite. Therefore, I've added a new test the applicable test classes to verify that the portIsFixed or not fixed depending on the configuration. I hope that's ok?